bulk_grades/views.py

```diff
@@ -15,37 +15,28 @@
 log = logging.getLogger(__name__)
 
 
-class GradeOnlyExportMixin(object):
+class GradeOnlyExport(View):
     """
     CSV Export of grade information only. To be used by both bulk grade export and interventions.
     """
 
+    def __init__(self, **kwargs):
+        """
+        Configure initial state.
+        """
+        super(GradeOnlyExport, self).__init__(**kwargs)
+        self.processor = None
+        self.operation_id = ''
+        self.filename = ''
+
     def _configure(self, request, course_id):
         """
         Dispatch django request.
         """
         if not (request.user.is_staff or request.user.has_perm('bulk_grades', course_id)):
             return HttpResponseForbidden('Not Staff')
         self.operation_id = request.GET.get('error_id', '')
-        if self.operation_id:
-            self.processor = api.GradeCSVProcessor.load(self.operation_id)
-            if self.processor.course_id != course_id:
-                return HttpResponseForbidden()
-        else:
-            assignment_grade_max = request.GET.get('assignmentGradeMax')
-            assignment_grade_min = request.GET.get('assignmentGradeMin')
-            self.processor = api.GradeCSVProcessor(
-                                                  course_id=course_id,
-                                                  _user=request.user,
-                                                  track=request.GET.get('track'),
-                                                  cohort=request.GET.get('cohort'),
-                                                  subsection=request.GET.get('assignment'),
-                                                  assignment_type=request.GET.get('assignmentType'),
-                                                  subsection_grade_max=(float(assignment_grade_max)
-                                                                        if assignment_grade_max else None),
-                                                  subsection_grade_min=(float(assignment_grade_min)
-                                                                        if assignment_grade_min else None),
-                                                )
+        self.initialize_processor(request, course_id)
```

Comment:
The `_configure()` method seems redundant, especially since the work has now moved to `initialize_processor`. Moving this stuff back to `dispatch` would be clearer.

Reply:
even though we would have duplicate lines ?

Reply:
What would be duplicated?

Reply:
the body of the _configure method